### PR TITLE
Pronouns & custom pronouns

### PR DESCRIPTION
**Pull request by DeadGirlRisen**
Bingle (dinofelisdruid) and I have worked on adding in the pronouns and custom pronouns. One of the people said we should submit it.

What this does:
initializes cats with pronouns, mostly using the preexisting code.
Makes sure genders are saved and can be loaded up.
Changes the specify gender button, it will now allow you to manually input custom pronouns.

With this change, cats can now have pronouns of a users choice and gender alignment, and the game will use these genders with the proper gender tagging in place.

### CODE REVIEW

File: scripts/cat/cats.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def __init__(self,
             self.adoptive_parents = []
             self.mate = []
             self.status = status
-            self.pronouns = [self.default_pronouns[0].copy()]
```
Comment (keyraven) [inline scripts/cat/cats.py:142]: This should be left. Every cat need to have pronouns, and in case there is an error with the assignment later, a cat should at least have one default pronoun.

File: scripts/cat/cats.py [hunk 2]
```diff
@@ -280,32 +280,41 @@ def __init__(self,
...
                     self.genderalign = "nonbinary"
                 else:
                     self.genderalign = self.gender
+                    self.pronouns = [self.default_pronouns[1].copy()]
             elif self.gender == "male" and not self.status in ['newborn', 'kitten']:
                 if trans_chance == 1:
                     self.genderalign = "trans female"
+                    self.pronouns = [self.default_pronouns[1].copy()]
                 elif nb_chance == 1:
                     self.genderalign = "nonbinary"
```
Comment (keyraven) [inline scripts/cat/cats.py]: There is no pronoun assignment for nonbinary cats.

File: scripts/game_structure/load_cat.py [hunk 3]
```diff
@@ -121,6 +121,7 @@ def json_load():
             new_cat.adoptive_parents = cat["adoptive_parents"] if "adoptive_parents" in cat else []
             
             new_cat.genderalign = cat["gender_align"]
+            new_cat.pronouns = cat["pronouns"]
```
Comment (keyraven) [inline scripts/game_structure/load_cat.py]: This causes a load-error when for older saves.  To fix this issue, use `.get` , with an appropriate default.

File: scripts/game_structure/windows.py [hunk 4]
```diff
@@ -462,14 +462,14 @@ class SpecifyCatGender(UIWindow):
     """This window allows the user to specify the cat's gender"""
 
     def __init__(self, cat):
-        super().__init__(scale(pygame.Rect((600, 430), (800, 370))),
+        super().__init__(scale(pygame.Rect((560, 360), (845, 470))),
```
Comment (keyraven) [inline scripts/game_structure/windows.py]: Regarding UI - I think this UI is too simple, and not very user friendly. There are no labels on the pronoun entry boxes, which may be confusing. In addition, there is no way to select from default pronouns, nor a way to add multiple pronouns.

**Review by keyraven — CHANGES_REQUESTED** [review]
In addition to my other comments, I am wary of pushing pronouns without a setting to refer to all cats as "they/them". That setting was promised, and we will have some upset players if the feature is pushed without the setting.

File: scripts/cat/cats.py (hunk 1 above)
Comment (DeadGirlRisen) [reply scripts/cat/cats.py:142]: Alright, I got this part added in, so I believe this comment can be resolved now. Working on the other stuff still with Bingle.

File: scripts/cat/cats.py (hunk 2 above)
Comment (DeadGirlRisen) [reply scripts/cat/cats.py]: Alright, I got this part added in, so I believe this comment can be resolved now. Working on the other stuff still with Bingle.

**Comment by DeadGirlRisen** [comment]
Alright! Changes have been addressed! Bingle and I have worked hard to get it all!

File: scripts/game_structure/windows.py (hunk 4 above)
Comment (DeadGirlRisen) [reply scripts/game_structure/windows.py]: We got it so you can select default pronouns, custom pronouns, have multiple pronouns. There's labels on the pronoun entry boxes. And an example box so you can test pronouns.

And we have a setting where it starts with all cats being they/them and you can turn it off.

Our customized pronouns is also an option to turn on or off, in case you just want the quick gender alignment.

File: scripts/game_structure/load_cat.py (hunk 3 above)
Comment (DeadGirlRisen) [reply scripts/game_structure/load_cat.py]: got this done! One of the easier things.

File: resources/dicts/events/ceremonies/ceremony_traits.json [hunk 5]
```diff
@@ -593,5 +593,23 @@
         "prudence",
         "judgement",
         "foresight"
+    ],
+    "cruel": [
```
Comment (keyraven) [inline resources/dicts/events/ceremonies/ceremony_traits.json]: Did you add a trait? If you wish to propose an additional trait, please remove and PR this separately

File: resources/dicts/names/names.json [hunk 6]
```diff
@@ -14,29 +14,26 @@
...
-    "roach", "roar", "rock", "root", "rose", "rumble", "rump", "run", "runner", "rush", "rustle", "scar", "scratch", "screech", 
-    "seed", "seeker", "shade", "shadow", "shard", "shell", "shimmer", "shine", "shiver", "shock", "shriek", "sight", "silk", "skip", 
-    "skitter", "sky", "slash", "slip", "snap", "snarl", "snout", "snow", "soar", "song", "spark", "speck", "speckle", "spider", 
-    "spike", "spirit", "splash", "splinter", "spore", "spot", "spots", "spring", "sprout", "stalk", "stem", "step", "sting", "stone", 
-    "storm", "streak", "stream", "strike", "stripe", "sun", "swarm", "swipe", "swoop", "talon", "tangle",
-    "thistle", "thorn", "thrift", "throat", "thrush", "thud", "thunder", "tiger", "timber", "toe", "tooth", "trail", "tree", 
-    "trot", "tumble", "twist", "valley", "watcher", "water", "wave", "web", "whisper", "whistle", "willow", "wind", "wing", "wish", "wisp", 
-    "zoom"
+    "arch", "ash", "aster", "back", "badger", "bark", "bat", "beak", "beam", "bee", "beetle", "bellow", "belly", "berry", "billow", "bird", "bite", "blaze", "blink", "bloom",
```
Comment (keyraven) [inline resources/dicts/names/names.json:17]: Why were changes made to the names? If you wish to propose additional names, please PR separately.

File: resources/dicts/patrols/general/medcat.json [hunk 7]
```diff
@@ -50,428 +65,1376 @@
                 ]
             },
             {
-                "text": "s_c makes sure everyone gets a good basking spot as the cats pause for a little break before resuming their herb-gathering.",
-                "exp": 20,
-                "weight": 20,
+                "text": "The sunlight feels great and s_c leads the cats in conversation, bringing them closer together.",
```
Comment (keyraven) [inline resources/dicts/patrols/general/medcat.json:68]: Please revert any added patrols. These need to be put in a separate PR.

File: resources/dicts/pronouns.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
+{
+    "default_pronouns": [
+        {
+            "name": "Neutral",
```
Comment (keyraven) [inline resources/dicts/pronouns.json:4]: Can you explain the purpose of the pronoun "name"?

File: scripts/cat/cats.py [hunk 9]
```diff
@@ -75,8 +75,10 @@ class Cat():
         "master": (321, 321)
     }
 
+
     default_pronouns = [
         {
+            "name": "Neutral",
```
Comment (keyraven) [inline scripts/cat/cats.py:81]: Are default pronouns defined here, or in the JSON resource file? These should only be defined once.

File: scripts/events_module/relationship/pregnancy_events.py [hunk 10]
```diff
@@ -765,8 +765,15 @@ def get_amount_of_kits(cat):
         three_kits = [min_kits + 2] * game.config["pregnancy"]["three_kit_possibility"][cat.age]
         four_kits = [min_kits + 3] * game.config["pregnancy"]["four_kit_possibility"][cat.age]
         five_kits = [min_kits + 4] * game.config["pregnancy"]["five_kit_possibility"][cat.age]
+        six_kits = [min_kits + 5] * game.config["pregnancy"]["six_kit_possibility"][cat.age]
+        seven_kits = [min_kits + 6] * game.config["pregnancy"]["seven_kit_possibility"][cat.age]
+        eight_kits = [min_kits + 7] * game.config["pregnancy"]["eight_kit_possibility"][cat.age]
+        nine_kits = [min_kits + 8] * game.config["pregnancy"]["nine_kit_possibility"][cat.age]
+        ten_kits = [min_kits + 9] * game.config["pregnancy"]["ten_kit_possibility"][cat.age]
+        eleven_kits = [min_kits + 10] * game.config["pregnancy"]["eleven_kit_possibility"][cat.age]
```
Comment (keyraven) [inline scripts/events_module/relationship/pregnancy_events.py:773]: We do not wish to increase the possible number of kits to 11. Please revert.

**Review by keyraven — CHANGES_REQUESTED** [review]
I am unable to access the pronoun change screen, so I can't give a proper review. 

Notes: 

- All the sprites have been replaced with a high-res texture pack, which currently render as black silhouettes.  Please restore the official spritesheets. 
- You also add a lot of thoughts, patrols, patrol art, skill paths. and an additional trait. Please revert these changes and PR them separately if you wish to propose these changes. 
- Unit tests and pronoun test are failing.

**Comment by DeadGirlRisen** [comment]
Ack! I'm so sorry! I don't know what happend?

It must have somehow pulled from my Lunar Mod?

I am so sorry!

Github is a bit confusing for me, so I must have messed up

**Review by DeadGirlRisen — COMMENTED** [review]
How do I undo this thing here?

**Comment by DeadGirlRisen** [comment]
Okay, So I don't know how to undo the accidental merge, but like, I know what files I messed with generally, so I'm going to create a clean branch again, put the files back, then it should be good to go. Im so sorry, I was trying to add this onto my lunar mod, but I must have somehow did this both ways?

**Comment by DeadGirlRisen** [comment]
Okay, I'm closing this, because I messed up, however I will reopen with a clean version, because im not sure how to fix this, im so sorry Keyraven for all the confusion and mess.
